core/arch/arch.c

```diff
@@ -3421,56 +3415,53 @@ priv_mcontext_to_dr_mcontext(dr_mcontext_t *dst, priv_mcontext_t *src)
      */
     if (dst->size > sizeof(dr_mcontext_t))
         return false;
-    if (TEST(DR_MC_ALL, dst->flags) &&
-        IF_X86_ELSE(dst->size > offsetof(priv_mcontext_t, opmask), true)) {
+    if (TEST(DR_MC_ALL, dst->flags) && dst->size == sizeof(dr_mcontext_t)) {
```

Comment:
bug: needs to be TESTALL as it originally was.

Reply:
Fixed.